src/Graphics/Datatypes/GLMVec.h

```diff
@@ -0,0 +1,63 @@
+/*
+   For more information, please see: http://software.sci.utah.edu
+
+   The MIT License
+
+   Copyright (c) 2020 Scientific Computing and Imaging Institute,
+   University of Utah.
+
+   Permission is hereby granted, free of charge, to any person obtaining a
+   copy of this software and associated documentation files (the "Software"),
+   to deal in the Software without restriction, including without limitation
+   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+   and/or sell copies of the Software, and to permit persons to whom the
+   Software is furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included
+   in all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+   OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+   THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+   DEALINGS IN THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <glm/glm.hpp>
+#include <glm/gtc/type_ptr.hpp>
+#include <Graphics/Datatypes/share.h>
+
+namespace SCIRun {
+namespace Graphics {
+  namespace Datatypes {
+    template <typename parent>
+    class GLMVec : public parent
+    {
+     public:
+      using parent::parent;
+
+      GLMVec(const parent& v) : parent(v) {}
+
+      float dot(const parent& other) const
+      {
+        return glm::dot(static_cast<parent>(*this), static_cast<parent>(other));
+      }
+
+      const parent& asParent() const { return static_cast<const parent&>(*this); }
+      const float* value_ptr() const { return glm::value_ptr(this->asParent()); }
+      float size() const { return this->asParent().length(); }
+      float length() const { return glm::length(this->asParent()); }
+      GLMVec<parent> normal() const { return glm::normalize(this->asParent()); }
+
+      GLMVec<parent> operator-(const GLMVec<parent>& other) { return glm::operator-(*this, other); }
+      GLMVec<parent> operator+(const GLMVec<parent>& other) { return glm::operator+(*this, other); }
+      GLMVec<parent> operator*(const GLMVec<parent>& other) { return glm::operator*(*this, other); }
+      GLMVec<parent> operator/(const GLMVec<parent>& other) { return glm::operator/(*this, other); }
```

Comment:
So the issue with rotation was happening because the math operators for this wrapper returned the parent class. Is this an appropriate way to overload those to return the wrapper instead?